hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncNonMetaRegionLocator.java

```diff
@@ -577,6 +624,15 @@ private void removeLocationFromCache(HRegionLocation loc) {
       if (!canUpdateOnError(loc, oldLoc)) {
         return;
       }
+      // Tell metaReplicaChooser that the location is stale. It will create a stale entry
+      // with timestamp internally. Next time the client looks up the same location,
+      // it will pick a different meta replica region. For the current implementation,
+      // the metaReplicaId is not used, so the primary one is passed in.
+      if (this.metaReplicaMode == CatalogReplicaMode.LoadBalance) {
```

Comment:
This sort of detail could be hidden inside the CatalogReplicaMode?

Reply:
hmmm... now I see CRM is just an enum.... but a CatalogReplica class that internally implements mode rather than scatter the processing here around locator? Can do later.

Reply:
Yeah, the mode is just an enum.